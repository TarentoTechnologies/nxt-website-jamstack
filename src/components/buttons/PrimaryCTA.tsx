import { Link } from "gatsby";
import React from "react";

import {
  labelStyles,
  large,
  primaryBtn,
  secondaryBtn,
} from "./PrimaryCTA.module.css";

interface PrimaryCTAProps {
  label: any;
  primary?: boolean;
  size?: string;
  clickHandler?: (event: any) => void;
  customClasses?: any;
  btnLink?: string;
  isMail?: boolean;
  isExternal?: boolean;
}

export const PrimaryCTA = ({
  label,
  primary = false,
  size,
  clickHandler,
  customClasses,
  btnLink,
  isMail,
  isExternal,
}: PrimaryCTAProps) => {
  const mode = primary ? `${primaryBtn}` : `${secondaryBtn}`;

  let link;
  if (btnLink != null) {
    if (isMail && !isExternal) {
      link = `mailto:${btnLink}`;
    } else {
      link = btnLink;
    }
  } else {
    link = "";
  }

  return (
    <>
      {isMail || isExternal ? (
        <a
          href={link}
          aria-label="link to specific detail page to know more"
          target="_blank"
          data-testid="primary-cta"
        >
          <div
            className={`text-center d-flex align-items-center justify-content-center 
      ${mode} ${size === "large" ? large : ""} ${customClasses}`}
            onClick={clickHandler}
            data-aos="fade-up"
          >
            <label className={`${labelStyles}`}>{label}</label>
          </div>
        </a>
      ) : (
        <Link to={btnLink != null ? btnLink : ""} data-testid="primary-cta">
          <div
            className={`text-center d-flex align-items-center justify-content-center 
      ${mode} ${size === "large" ? large : ""} ${customClasses}`}
            onClick={clickHandler}
            data-aos="fade-up"
          >
            <label className={`${labelStyles}`}>{label}</label>
          </div>
        </Link>
      )}
    </>
  );
};
