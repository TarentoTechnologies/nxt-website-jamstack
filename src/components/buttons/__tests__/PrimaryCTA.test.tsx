import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { PrimaryCTA } from "../PrimaryCTA";

describe("Test if the PrimaryCTA component is working as expected", () => {
  it("<PrimaryCTA /> matches snapshot", () => {
    const component = render(
      <PrimaryCTA data-testid="primary-cta" label="Register Now" primary />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<PrimaryCTA /> renders without exploding", () => {
    const component = render(
      <PrimaryCTA data-testid="primary-cta" label="Register Now" primary />
    );
    expect(component.getByTestId("primary-cta")).toBeInTheDocument();
  });
});
