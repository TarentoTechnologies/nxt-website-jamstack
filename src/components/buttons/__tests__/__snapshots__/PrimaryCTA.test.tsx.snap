// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the PrimaryCTA component is working as expected <PrimaryCTA /> matches snapshot 1`] = `
<div>
  <a
    data-testid="primary-cta"
    href=""
  >
    <div
      class="text-center d-flex align-items-center justify-content-center 
      primaryBtn  undefined"
      data-aos="fade-up"
    >
      <label
        class="labelStyles"
      >
        Register Now
      </label>
    </div>
  </a>
</div>
`;
